notebooks/py/assay_comparison.py

```diff
@@ -0,0 +1,237 @@
+spots(decoded_image=dartfish_decoded_image, background_image=dartfish_dots, decoded_image_kwargs=dict(alpha=1.), ax=ax4)
+spots(decoded_image=dartfish_decoded_image, background_image=dartfish_nuclei, decoded_image_kwargs=dict(alpha=1.), ax=ax6);
+# EPY: END code
+
+# EPY: START markdown
+### Show different background types for ISS
+# EPY: END markdown
+
+# EPY: START code
+f, ((ax1, ax2), (ax3, ax4), (ax5, ax6)) = plt.subplots(ncols=2, nrows=3, figsize=(30, 40))
+spots(iss, background_image=np.zeros_like(iss_dots), spots_kwargs=dict(alpha=1.), ax=ax1)
+spots(iss, background_image=iss_dots, spots_kwargs=dict(alpha=1.), ax=ax3)
+spots(iss, background_image=iss_nuclei, spots_kwargs=dict(alpha=1.), ax=ax5)
+
+# ISS doesn't have a decoded image, so leave those plots blank to make this clear
+for ax in (ax2, ax4, ax6):
+    ax.set_axis_off()
+f.tight_layout()
+# EPY: END code
+
+# EPY: START markdown
+### Download available copy number information from original authors
+# EPY: END markdown
+
+# EPY: START code
+dartfish_copy_number = pd.read_csv('https://dmf0bdeheu4zf.cloudfront.net/20180821/DARTFISH/fov_001/counts.csv', index_col=0, squeeze=True)
+merfish_copy_number = pd.read_csv(data_root + "merfish_copy_number_benchmark.csv", index_col=0, squeeze=True)
+iss_copy_number = pd.read_csv(data_root + "iss_copy_number_benchmark.csv", index_col=1)['cnt']
+# EPY: END code
+
+# EPY: START code
+f, axes = plt.subplots(ncols=3, nrows=4, figsize=(12, 14))
+
+for dataset, benchmark, axis in zip([iss, merfish, dartfish], [iss_copy_number, merfish_copy_number, dartfish_copy_number], axes[0, :]):
+    compare_copy_number(dataset, benchmark, ax=axis, color='tab:blue')
+
+for dataset, axis in zip(datasets, axes[1, :]):
+    barcode_magnitude_histogram(dataset, bins=20, ax=axis)
+
+for dataset, axis in zip(datasets, axes[2, :]):
+    spot_area_histogram(dataset, bins=20, ax=axis)
+
+for dataset, axis in zip([merfish, dartfish], axes[3, 1:]):
+    spot_distance_histogram(dataset, bins=20, ax=axis)
+
+# set the assay names as the titles of the top plots
+axes[0, 0].set_title('In-Situ Sequencing', fontsize=20)
+axes[0, 1].set_title('MERFISH', fontsize=20)
+axes[0, 2].set_title('DARTFISH', fontsize=20);
+
+# reset y-axis labels
+for ax in np.ravel(axes):
+    ax.set_ylabel('')
+    
+# reset titles
+for ax in np.ravel(axes[1:, :]):
+    ax.set_title('')
+
+# set the y-axis labels
+column_1_axes = (axes[:, 0])
+plot_names = (
+    'Copy Number Comparison\nwith Author Pipelines\n\nstarfish result\n\n',
+    'Barcode Magnitude\nDistributions\n\nnumber of pixels',
+    'Spot Area\nDistributions\n\nnumber of spots',
+    'Feature Distances to\nNearest Code\n\nnumber of features\n\n'
+)
+for ax, name in zip(column_1_axes, plot_names):
+    ax.set_ylabel(name, fontsize=16)
+
+
+# fix up the figure
+f.tight_layout()
+for ax in np.ravel(axes):
+    sns.despine(ax=ax)
+
+# turn off the empty plots, matplotlib is super bad about this. 
+for ax in (axes[3, 0], axes[0, 0]):
```

Comment:
Why is there a plot, but no axes for the Copy number comparison - ISS plot? Maybe the seaborn plot on the ax isn't a patch and thus cannot be toggled off with ax.patch.set_visible()?

Reply:
Oops. I originally didn't have a copy number plot, and I forgot to remove this. Nice catch!